code/modules/research/xenobiology/crossbreeding/_status_effects.dm

```diff
@@ -965,15 +965,22 @@
 /datum/status_effect/stabilized/rainbow
 	id = "stabilizedrainbow"
 	colour = "rainbow"
+	var/messagefired = FALSE
 
 /datum/status_effect/stabilized/rainbow/tick()
-	if(owner.health <= 0)
+	if(owner.health <= 0 && !messagefired)
 		var/obj/item/slimecross/stabilized/rainbow/X = linked_extract
 		if(istype(X))
 			if(X.regencore)
-				X.regencore.afterattack(owner,owner,TRUE)
-				X.regencore = null
 				owner.visible_message("<span class='warning'>[owner] flashes a rainbow of colors, and [owner.p_their()] skin is coated in a milky regenerative goo!</span>")
-				qdel(src)
-				qdel(linked_extract)
+				addtimer(CALLBACK(src, .proc/activate), 50)
```

Comment:
This isn't a do_after, this is a 5 second timer. Using do_after would mean you need to stay still for 5 seconds to use it and remain concious, this means that once triggered the effect will activate after 5 seconds no matter what you do, as long as you could activiate it in the first place